apps/test/unit/code-studio/pd/workshop_enrollment/enroll_formTest.js

```diff
@@ -146,8 +146,8 @@ describe('Enroll Form', () => {
       enrollForm.find('#submit').simulate('click');
 
       expect(jQuery.ajax.calledOnce).to.be.true;
-      expect(jQuery.ajax.getCall(0).args[0].data).to.equal(
-        JSON.stringify(expectedData)
+      expect(JSON.parse(jQuery.ajax.getCall(0).args[0].data)).to.deep.equal(
+        expectedData
       );
```

Comment:
I'm curious it is about the upgrade that broke this test.

Reply:
The symptom was that the expected and actual JSON had the same contents, but did not match because the order of the keys was different.

we switched from `transform-object-rest-spread` to `@babel/plugin-proposal-object-rest-spread` here: https://github.com/code-dot-org/code-dot-org/pull/32266/commits/b122989214021d6d9dec274a0b8f4ce20786d313#diff-48b49bb7e2aefb54ba92f64f2efffcffR7 while most of the changes in that commit are simply moving from an older version to a renamed version of the same library, this particular line represents a move to a different object rest spread implementation, so my assumption is that the new implementation produces objects with keys in different order.

doing a deep copy makes the comparison work regardless of the order of the keys.

Reply:
Awesome, thanks for tracking that down!